pkg/objstore/azure/helpers.go

```diff
@@ -68,6 +66,23 @@ func getAzureStorageCredentials(logger log.Logger, conf Config) (blob.Credential
 	return credential, nil
 }
 
+func getServicePrincipalToken(logger log.Logger, conf Config) (*adal.ServicePrincipalToken, error) {
+	resource := conf.MSIResource
+	if resource == "" {
+		resource = fmt.Sprintf("https://%s.%s", conf.StorageAccountName, conf.Endpoint)
+	}
+
+	if conf.UserAssignedID != "" {
+		level.Debug(logger).Log("msg", "using user assigned identity:", conf.UserAssignedID)
+		return adal.NewServicePrincipalTokenFromMSIWithUserAssignedID("", resource, conf.UserAssignedID)
```

Comment:
Nit: This function is deprecated, instead `NewServicePrincipalTokenFromManagedIdentity` should be used.
https://github.com/Azure/go-autorest/blob/cf74520d3483ff497328aa9f0f0e2ed5e55a70ad/autorest/adal/token.go#L749-L752

Reply:
Thanks for the note. Updated to not use the deprecated method. I ended up using the same constructs as the system-assigned identity existing code, and only set the client id if the user assigned identity is specified, which further simplified the code.